image_tools_py/src/cam2image_py.py

```diff
@@ -0,0 +1,183 @@
+        help='History QoS setting, 0 - keep last sample, 1 - keep all the samples')
+    parser.add_argument(
+        '-r', '--reliability', dest='reliability_policy', action='store', default=1, type=int,
+        choices=[0, 1],
+        help='Reliability QoS setting, 0 - best effort, 1 - reliable')
+    parser.add_argument(
+        '-s', '--show', dest='show_camera', action='store', default=0, type=int, choices=[0, 1],
+        help='Show the camera stream')
+    parser.add_argument(
+        '-x', '--width', dest='width', action='store', default=320, type=int,
+        help='Image width')
+    parser.add_argument(
+        '-y', '--height', dest='height', action='store', default=240, type=int,
+        help='Image height')
+    args = parser.parse_args()
+
+    # Initialize a ROS 2 node to publish images read from the OpenCV interface to
+    # the camera.
+    node = rclpy.create_node('cam2imagepy')
+
+    # Set the parameters of the quality of service profile.  Initialize as the
+    # default profile and set the QoS parameters specified on the command line.
+    custom_camera_qos_profile = image_qos_profile.create_qos_profile(
+        args.queue_depth, args.reliability_policy, args.keep)
+
+    # Create the image publisher with our custom QoS profile.
+    pub = node.create_publisher(sensor_msgs.msg.Image, 'image', custom_camera_qos_profile)
+
+    custom_flip_qos_profile = qos_profile_sensor_data
+    custom_flip_qos_profile.depth = 10
+    node.create_subscription(
+        std_msgs.msg.Bool, 'flip_image', flip_image_cb, custom_flip_qos_profile)
+
+    if args.burger_mode:
+        burger_cap = burger_py.Burger()
+    else:
+        # Initialize OpenCV video capture stream.  Always open device 0.
+        cam_cap = cv2.VideoCapture(0)
+
+        # Set the width and height based on command-line arguments.
+        cam_cap.set(cv2.CAP_PROP_FRAME_WIDTH, args.width)
+        cam_cap.set(cv2.CAP_PROP_FRAME_HEIGHT, args.height)
+        if not cam_cap.isOpened():
+            print('Could not open video stream')
+            sys.exit(1)
+
+    # Our main event loop will spin until the user presses CTRL-C to exit.
+    frame_number = 1
+    while rclpy.ok():
+        # Get the frame from the video capture.
+        frame = None
+        if args.burger_mode:
+            frame = burger_cap.render_burger(args.width, args.height)
+        else:
+            ret, frame = cam_cap.read()
+
+        # Check if the frame was grabbed correctly.
+        if frame is not None:
+            global is_flipped
+            # Convert to a ROS image
+            if is_flipped:
+                # Flip the frame if needed
+                flipped_frame = cv2.flip(frame, 1)
+                msg = convert_frame_to_message(flipped_frame, frame_number)
+            else:
+                msg = convert_frame_to_message(frame, frame_number)
+
+            if args.show_camera == 1:
```

Comment:
If show_camera was a boolean, I would have done it that way.  Given that it is an integer, I prefer to be explicit (even though you are correct in that it would work).